sdcm/wait.py

```diff
@@ -53,18 +53,27 @@ def retry_logger(retry_state):
             retry_state.attempt_number,
             retry_state.outcome._exception or retry_state.outcome._result,
         )
+    stops = [tenacity.stop_after_delay(timeout)]
+    if stop_event := kwargs.pop("stop_event", None):
+        stops.append(tenacity.stop.stop_when_event_set(stop_event))
 
     try:
         retry = tenacity.Retrying(
             reraise=throw_exc,
-            stop=tenacity.stop_after_delay(timeout),
+            stop=tenacity.stop_any(*stops),
             wait=tenacity.wait_fixed(step),
             before_sleep=retry_logger,
             retry=(tenacity.retry_if_result(lambda value: not value) | tenacity.retry_if_exception_type())
         )
         res = retry(func, **kwargs)
 
     except Exception as ex:  # pylint: disable=broad-except
+        if stop_event and stop_event.is_set():
```

Comment:
2dt unit test for this, for different combinations for timeout(exceed/nor exceed) , event (set/not set), func result (result/exception)

Reply:
done

Reply:
not covered:  func result (result/exception)

I think we need cover logic when there is an exception and stop_event  set/not set

Reply:
What kind of exception: by timeout  or exception in target function?

Reply:
- test_04_return_result_before_stop_event_and_wait_timeout - return target function result before event was set and timeout
- test_04_raise_by_timeout_before_set_event  - raise by timeout before event set

Reply:
exception in target function and  event  set
exception in target function and  event is not  set

Reply:
Added. But due to we retry if any exception triggered, the behavior was not changed